bin/stalePullRequest.js

```diff
@@ -1,92 +1,131 @@
+    dateLog('Checking ' + repositoryName);
+    var pullRequests = [];
+
+    function processPage(response) {
+        var linkData = parseLink(response.headers.link);
+        pullRequests = pullRequests.concat(response.body);
+        // If we're at the last page
+        if (!Cesium.defined(linkData) || !Cesium.defined(linkData.next)) {
+            return Promise.each(pullRequests, function (pullRequest) {
+                return stalePullRequest._processPullRequest(pullRequest, repositorySettings);
+            });
+        }
+        // Otherwise, request the next page
+        return requestPromise.get({
+            url: linkData.next.url,
+            headers: repositorySettings.headers,
+            json: true,
+            resolveWithFullResponse: true
+        }).then(processPage);
+    }
+
+    return requestPromise.get({
+        url: 'https://api.github.com/repos/' + repositoryName + '/pulls?state=open&base=master',
+        headers: repositorySettings.headers,
+        json: true,
+        resolveWithFullResponse: true
+    }).then(processPage);
+};
+
+stalePullRequest._processPullRequest = function (pullRequest, repositorySettings) {
+    var commentsUrl = pullRequest.comments_url;
+    return requestPromise.get({
+        url: commentsUrl,
+        headers: repositorySettings.headers,
+        json: true,
+        resolveWithFullResponse: true
+    })
+        .then(function (response) {
+            var linkData = parseLink(response.headers.link);
+            if (Cesium.defined(linkData)) {
+                commentsUrl = linkData.last.url;
+            }
+            return requestPromise.get({
+                url: commentsUrl,
+                headers: repositorySettings.headers,
+                json: true,
+            }).then(function (commentsJsonResponse) {
+                var lastComment = commentsJsonResponse[commentsJsonResponse.length - 1];
+                var foundStop = stalePullRequest.foundStopComment(commentsJsonResponse);
+                if (!foundStop && stalePullRequest.daysSince(new Date(lastComment.created_at)) >= repositorySettings.maxDaysSinceUpdate) {
+                    var template = repositorySettings.stalePullRequestTemplate;
+                    return requestPromise.post({
+                        url: commentsUrl,
+                        headers: repositorySettings.headers,
+                        body: {
+                            body: template({
+                                maxDaysSinceUpdate: repositorySettings.maxDaysSinceUpdate
+                            })
+                        },
+                        json: true
+                    });
+                }
+            });
+        });
+};
+
+stalePullRequest.daysSince = function (date) {
+    var msPerDay = 24 * 60 * 60 * 1000;
+    return (Date.now() - date.getTime()) / msPerDay;
+};
+
+stalePullRequest.foundStopComment = function (commentsJsonResponse) {

```

Comment:
I don't think this function is tested at all. Try running the coverage task.

Reply:
Added a test for it! Coverage looks a lot better now.